realsense2_description/urdf/_d405.urdf.xacro

```diff
@@ -0,0 +1,165 @@
+<?xml version="1.0"?>
+
+<!--
+# Copyright 2023 Intel Corporation. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+-->
+
+<!--
+This is the URDF model for the Intel RealSense 405 camera, in its
+aluminum peripherial evaluation case.
+-->
+
+<robot name="sensor_d405" xmlns:xacro="http://ros.org/wiki/xacro">
+  <!-- Includes -->
+  <xacro:include filename="$(find realsense2_description)/urdf/_materials.urdf.xacro" />
+  <xacro:include filename="$(find realsense2_description)/urdf/_usb_plug.urdf.xacro" />
+
+  <xacro:macro name="sensor_d405" params="parent *origin name:=camera use_nominal_extrinsics:=false">
+    <xacro:arg name="add_plug" default="false" />
+    <xacro:property name="M_PI" value="3.1415926535897931" />
+
+    <!-- The following values are approximate, and the camera node
+     publishing TF values with actual calibrated camera extrinsic values -->
+    <xacro:property name="d405_cam_depth_to_infra1_offset" value="0.0"/>
+    <xacro:property name="d405_cam_depth_to_infra2_offset" value="-0.018"/>
+    <xacro:property name="d405_cam_depth_to_color_offset" value="0.0"/>
+
+    <!-- The following values model the aluminum peripherial case for the
+  	d405 camera, with the camera joint represented by the actual
+  	peripherial camera tripod mount -->
+    <xacro:property name="d405_cam_width" value="0.042"/>
```

Comment:
Just curios , this is generated automatically some how or you set this values according to the STL file manually?

Reply:
These values are set manually based on the dimensions mentioned in the datasheet.